src/test/scala/com/eclipsesource/schema/FormatSpec.scala

```diff
@@ -96,15 +117,46 @@ class FormatSpec extends Specification with JsonSpec {
         }
       }
       val validatorWithRangeFormat = validator.addFormat(rangeFormat)
-      val schema = JsonSource.schemaFromString(
+      val numberSchema = JsonSource.schemaFromString(
         """
           |{
           |  "type": "number",
           |  "format": "range0To10"
           | }
         """.stripMargin).get
-      validatorWithRangeFormat.validate(schema, JsNumber(10)).isSuccess must beTrue
-      validatorWithRangeFormat.validate(schema, JsNumber(-1)).isError must beTrue
+
+      val integerSchema = JsonSource.schemaFromString(
+        """
+          |{
+          |  "type": "integer",
+          |  "format": "range0To10"
+          | }
+        """.stripMargin).get
+
+      validatorWithRangeFormat.validate(integerSchema, JsNumber(10)).isSuccess must beTrue
+      validatorWithRangeFormat.validate(integerSchema, JsNumber(-1)).isError must beTrue
+
+      validatorWithRangeFormat.validate(numberSchema, JsNumber(10)).isSuccess must beTrue
+      validatorWithRangeFormat.validate(numberSchema, JsNumber(-1)).isError must beTrue
+    }
+
+    "validate integer against double max boundary" in {
+      val schema = JsonSource.schemaFromString(
+        """
+          |{
+          |  "type" : "object",
+          |  "properties" : {
+          |    "intprop" : {
+          |      "type" : "integer",
+          |      "format" : "int32"
+          |    }
+          |  }
+          |}
+        """.stripMargin).get
+      val instance = Json.obj(
+        "intprop" -> -1.7976931348623157E+308
```

Comment:
Yes, it passed and still does if you leave out the `format`, but as previously mentioned, I'll add an explicit test.